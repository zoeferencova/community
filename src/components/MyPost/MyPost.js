import React, { Component } from "react";
import styles from "./MyPost.module.css";

export default class MyPost extends Component {
    capitalizeFirstLetter(string) {
        return string.charAt(0).toUpperCase() + string.slice(1);
    }
    render() {
        return (   
            <p className={styles.userPost}>{this.capitalizeFirstLetter(this.props.type)} created {Date.now() - this.props.timestamp} hours ago</p>
        )
    }
}