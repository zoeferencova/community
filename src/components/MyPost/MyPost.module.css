.userPost {
    padding: 2px 0px;
    margin: 5px 0px;
    font-size: 14px;
    text-align: left;
}

.link {
    color: inherit;
    text-decoration: none;
}

.heart,
.hand {
    margin-right: 7px;
    opacity: 0.7;
}

.heart {
    color: #bf5e6b;
}

.hand {
    color: #e6c78e;
}