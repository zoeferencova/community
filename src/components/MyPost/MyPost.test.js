import React from "react";
import { render } from "enzyme";
import MyPost from "./MyPost";
import { BrowserRouter } from "react-router-dom";

describe("MyPost", () => {
  it("renders MyPost component", () => {
    const mockProps = {
        date_created: "2020-09-23T20:33:19.724Z",
        id: 40,
        post_type: "request"
    }

    const wrapper = render(
      <BrowserRouter>
        <MyPost {...mockProps} />
      </BrowserRouter>
    );
    
    expect(wrapper).toMatchSnapshot();
  })
})



