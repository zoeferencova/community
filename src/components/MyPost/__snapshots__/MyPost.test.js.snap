// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPost renders MyPost component 1`] = `
<a
  class="link"
  href="/my-post/40"
>
  <p
    class="userPost"
  >
    <i
      class="fas fa-hand-paper hand"
    />
     Request created 12 days ago
  </p>
</a>
`;
